framework/src/main/java/org/apache/drill/test/framework/TestVerifier.java

```diff
@@ -518,6 +520,30 @@ private static boolean matchesAll(String actual, String expected) {
     return true;
   }
 
+  /**
+   * Check for regex matches.  The patterns can appear in any order in the
+   *  expected results file.
+   * 
+   * @param actual
+   *          string containing the actual results
+   * @param expected
+   *          string containing the expected results
+   * @return true or false to indicate if there is a match
+   */
+  private static boolean matchesAllNoOrder(String actual, String expected) {
+    String[] expectedLines = expected.split("\n");
+    actual = actual.trim();
+    int i = 0;
+    for (String string : expectedLines) {
+      string = string.trim();
+      Matcher matcher = Pattern.compile(string).matcher(actual);
+      if (!matcher.find()) {
```

Comment:
I think this misses one case, where the result has more partitions than needed.  If the query should scan 3 partitions, but ends up scanning 4 partitions, I don't see a way to catch this with the new code.  So I think we need different methods depending on the use case.  I think it is hard to create one method that will work in all cases.

With the original code and the original use case (parquet filter pushdown), we check for extra partitions, missing partitions, and repeated partitions.